tests/components/volvo/test_coordinator.py

```diff
@@ -0,0 +1,94 @@
+"""Test Volvo coordinator."""
+
+from unittest.mock import AsyncMock
+
+import pytest
+from volvocarsapi.api import VolvoCarsApi
+from volvocarsapi.models import VolvoApiException, VolvoAuthException
+
+from homeassistant.components.volvo.coordinator import VolvoDataCoordinator
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, HomeAssistantError
+from homeassistant.helpers.update_coordinator import UpdateFailed
+
+from tests.common import MockConfigEntry
+
+
+async def test_setup_coordinator_auth_failure(
```

Comment:
We normally avoid testing internal things like the coordinator own methods but instead just asserting entity state etc when a coordinator fails to setup or refresh.
You can take a look at `nordpool` for a simple example.